api/src/main/java/org/openmrs/module/addresshierarchy/config/ConfigLoaderUtil.java

```diff
@@ -0,0 +1,405 @@
+	 *            "subdir/config.xml"
+	 * @return An empty string if the checksum hasn't changed, the new checksum otherwise.
+	 */
+	public static String getChecksumIfChanged(String configDirPath, String configFilePath) {
+		String savedChecksum = readLatestChecksum(configDirPath, configFilePath);
+		String checksum = computeChecksum(configDirPath, configFilePath);
+		return savedChecksum.equals(checksum) ? "" : checksum;
+	}
+	
+	/**
+	 * @see #getChecksumIfChanged(String, String)
+	 */
+	public String getChecksumIfChanged(String configFilePath) {
+		return getChecksumIfChanged(configDirPath, configFilePath);
+	}
+	
+	/**
+	 * Fetches all the files in a directory based on their extension.
+	 * 
+	 * @param configDirPath The absolute path to the config. directory, eg.
+	 *            "../configuration/addresshierarchy".
+	 * @param extension The extension to filter for, eg "xml".
+	 * @param recursive Set to true to continue recursively into subdirectories.
+	 * @return The list of {@link File} instances.
+	 */
+	public static List<File> getFiles(String configDirPath, String extension, boolean recursive) {
+		
+		final List<File> allFiles = new ArrayList<File>();
+		
+		final File[] files = new File(configDirPath).listFiles(getExtensionFilenameFilter(extension));
+		if (files != null) {
+			allFiles.addAll(Arrays.asList(files));
+		}
+		
+		if (recursive) {
+			final File[] dirs = new File(configDirPath).listFiles(getDirectoryFilenameFilter());
+			if (dirs != null) {
+				for (File dir : dirs) {
+					allFiles.addAll(getFiles(dir.getPath(), extension, true));
+				}
+			}
+		}
+		
+		return allFiles;
+	}
+	
+	/**
+	 * @see #getFiles(String, String, boolean)
+	 */
+	public static List<File> getFiles(String configDirPath, String extension) {
+		return getFiles(configDirPath, extension, false);
+	}
+	
+	/**
+	 * @see #getFiles(String, String, boolean)
+	 */
+	public List<File> getFiles(String extension, boolean recursive) {
+		return getFiles(configDirPath, extension, recursive);
+	}
+	
+	/**
+	 * @see #getFiles(String, String)
+	 */
+	public List<File> getFiles(String extension) {
+		return getFiles(configDirPath, extension);
+	}
+	
+	/**
+	 * Fetches the config. file from its relative path inside the configuration folder.
+	 * 
+	 * @param configDirPath The absolute path to the containing directory, eg.
+	 *            "../configuration/addresshierarchy"
+	 * @param configFilePath The config. file relative path inside the containing directory, eg.
+	 *            "subdir/config.xml"
+	 * @return The {@link File} instance.
+	 */
+	public static File getFile(String configDirPath, String configFilePath) {
+		StringBuilder path = new StringBuilder(configDirPath);
+		path.append(File.separator).append(configFilePath);
+		return new File(path.toString());
+	}
+	
+	/**
+	 * @see #getFile(String, String)
+	 */
+	public File getFile(String configFilePath) {
+		return getFile(configDirPath, configFilePath);
+	}
+	
```

Comment:
By the way, I think we will change the behavior of `ConfigLoaderUtil` so that the checksums are not saved anymore alongside the config files themselves. Mixing them as proven to be slightly challenging when pushing updated configs while working on a specific implementation's custom configuration.
I know that you would like to avoid bloating the app data dir but I was simply thinking of a mirrored structure for the checksums, like this:
```
/opt/openmrs
  \_ configuration/
  \_ configuration_checksums/
```
Someone could just hit `tree` to quickly see what was ever processed or not.